cypress/pages/BounceRulesContainer/index.js

```diff
@@ -157,9 +157,22 @@ class BounceRulesPage extends Page {
     return this.confirmationSubmit.click();
   }
 
-  deleteBounceRuleUI(id) {
-    this.createdRuleButton(id).click();
-    return this.deleteConfirmationConfirm.click();
+  deleteBounceRuleUI(testRule) {
+    cy.task("getRules").then(res => {
+      for (let i = 0; i < res.length; i++) {
```

Comment:
Same here let's change all for loops to use the array helpers since they're functional and easier to understand the intent. We can do stuff with arr.find and arr.some and rearrange this to be a lot shorter and descriptive